WooCommerce/StoreWidgets/StoreInfoProvider.swift

```diff
@@ -1,13 +1,25 @@
 import WidgetKit
 import KeychainAccess
 
-/// Type that represents the Widget information
+/// Type that represents the all the possible Widget states.
 ///
-struct StoreInfoEntry: TimelineEntry {
-    /// Date to request new info
-    ///
-    var date: Date
+enum StoreInfoEntry: TimelineEntry {
+    // Represents a not logged-in state
+    case notConnected
+
+    // Represents a fetching error state
+    case error
+
+    // Represents a fetched data state
+    case data(StoreInfoData)
 
+    // Current date, needed by the `TimelineEntry` protocol.
+    var date: Date { Date() }
```

Comment:
This doesn't look correct. `TimelineEntry` captures widget state for a single particular date. With this implementation they all will look like an entry for most current datetime. Can it break refresh timeout or something else?
I don't see any issues now, but I haven't tested actual background widget refresh after 30 mins.

Reply:
To my understanding, this date is used when you provide multiple entries in the `getTimeline` method, so `WidgetKit` knows when to display each of the entries. 

The refresh date is not compared against any of the entry dates. There, we provide a fixed point in time where the refresh is executed.

So having these entries with today's date should be safe and expected.